rsciio/tests/test_topspin.py

```diff
@@ -0,0 +1,170 @@
+    assert out_C[1]["data"].shape == (3, 5, 16, 16)
+    assert out_C[2]["data"].shape == (11, 13)
+
+    # Check identical data loaded with the subset call are identical
+    assert np.all(sub_b1a[0]["data"] == sub_b1[0]["data"])
+    assert np.all(sub_b1a[0]["data"] == out_B[0]["data"])
+
+    # Check hyperspy metadata is populated for all test datasets
+    for out in [out_A, out_B, out_C, sub_b1, sub_b1a]:
+        for dset in out:
+            md = dset["metadata"]
+            assert "General" in md
+            assert "title" in md["General"]
+            assert isinstance(md["General"]["FileIO"], dict)
+            for key in md.keys():
+                assert md[key] is not None  # exists
+                if isinstance(md[key], str):
+                    assert len(md[key]) > 1  # has non-default data
+
+    # Check axes
+    for out in [out_A, out_B, out_C, sub_b1, sub_b1a]:
+        for dset in out:
+            ad_all = dset["axes"]
+            for ad in ad_all:
+                assert "name" in ad.keys()
+                assert "units" in ad.keys()
+                assert "size" in ad.keys()
+                assert "scale" in ad.keys()
+                assert "offset" in ad.keys()
+                assert "navigate" in ad.keys()
+                assert "index_in_array" in ad.keys()
+                for k in ["size", "scale", "offset"]:
+                    assert np.abs(ad[k]) > 0
+                if ad["name"] in ["x", "y"]:
+                    assert ad["navigate"] == True
+                else:
+                    assert ad["navigate"] == False
+            names = np.array([x["name"] for x in ad_all])
+            idxs = np.array([x["index_in_array"] for x in ad_all])
+            assert np.all(np.isin(np.unique(names), ["x", "y", "kx", "ky"]))
+            assert np.max(np.unique(names, return_counts=True)[1]) == 1
+            assert np.max(np.unique(idxs, return_counts=True)[1]) == 1
+
+
+def test_dryrun():
+    correct_sizes = [
+        ["11, 13", "2, 5, 16, 16"],
+        ["3, 7, 37, 37", "11, 13", "11, 13", "11, 13"],
+        ["11, 13", "3, 5, 16, 16", "11, 13"],
+    ]
+    for i, f in enumerate([file_A, file_B, file_C]):
+        buffer = StringIO()
+        with redirect_stdout(buffer):
+            out = file_reader(f, dryrun=True)
+        assert out == []
+        txt = buffer.getvalue()
+        # make sure the output is correctly estimating the object shapes
+        dims_str = [x.split("]")[0] for x in txt.split("[")[1:]]
+        assert dims_str == correct_sizes[i]
+
+
+def test_with_hyperspy():
+    hs = pytest.importorskip("hyperspy.api", reason="hyperspy not installed")
+    for f in [file_A, file_B, file_C]:
+        out = file_reader(f, show_progressbar=False)
+        for dset in out:
+            signal = hs.signals.Signal2D(
+                data=dset["data"], axes=dset["axes"], metadata=dset["metadata"]
+            )
+            assert isinstance(signal, hs.signals.Signal2D)
```

Comment:
Can you check the value of the `scale`, `offsets` and `units` by using the value shown Topspin software? These are quite important and needs to be tested.